### ISSUE TITLE
[Feature Request]: Tag autocomplete when adding tags

### ISSUE THREAD

**Issue by Tyrannicodin** [TagStudioDev/TagStudio#476]
### Checklist

- [X] I am using an up-to-date version.
- [X] I have read the [documentation](https://github.com/TagStudioDev/TagStudio/blob/main/docs/index.md).
- [X] I have searched existing [issues](https://github.com/TagStudioDev/TagStudio/issues).

### Description

This would add a menu that tries to autocomplete the tag the user is trying to add via a text box, similar to hydrus web:
![image](https://github.com/user-attachments/assets/d6622f9e-91e5-4f7f-bced-8427f6b7a551)

This also includes an add button if users would prefer to add via a context menu/modal.

**Comment by python357-1** [comment]
@Tyrannicodin do you mind if i take a crack at this?

**Comment by Tyrannicodin** [comment]
Not at all, I've been pretty busy lately and have been putting most of my time into other projects so I'd really appreciate it.

**Comment by ToxicMushroom** [comment]
I don't see what benefit this has over the current popup ?

**Comment by CyanVoxel** [comment]
This feature was requested before #655 streamlined the process for adding tags, so yeah there should no longer be a need for this feature given that the current process for adding tags provides the same information with the same number of key/mouse presses.
